components/brave_vpn/brave_vpn_service.cc

```diff
@@ -870,10 +870,37 @@ void BraveVpnService::OnSuspend() {
   // Set reconnection state in case if computer/laptop is going to sleep.
   // The disconnection event will be fired after waking up and we want to
   // restore the connection.
-  needs_connect_ = is_connected();
+  if (is_connected()) {
+    Disconnect();
+    reconnect_on_resume_ = true;
+  }
+  VLOG(2) << __func__
+          << " Should reconnect when resume:" << reconnect_on_resume_;
+}
+
+void BraveVpnService::OnResume() {
+#if BUILDFLAG(IS_MAC)
+  OnDNSChanged();
+#endif  // BUILDFLAG(IS_MAC)
+}
+
+void BraveVpnService::OnDNSChanged() {
+  if (!IsNetworkAvailable() ||
+      // This event is triggered before going to sleep while vpn is still
```

Comment:
nti: I think comments should be located before the `if clause` instead of between each condition.

Reply:
This comment explains why we add that second condition, will change in next PR